### PR TITLE
Fix bug in HTML UI due to upgrade to Jena 3

### PR DESCRIPTION
**Pull request by awoods**
Resolves: https://jira.duraspace.org/browse/FCREPO-2124

Related to: https://jira.duraspace.org/browse/FCREPO-1672

### CODE REVIEW

File: fcrepo-http-api/src/main/resources/views/common.vsl [hunk 1]
```diff
@@ -46,7 +46,7 @@
 #end
 
 #macro( isObject $sub )
-#set($type = $anynode.createURI("http://fedora.info/definitions/v4/repository#mixinTypes"))
+#set($type = $helpers.createURI("http://fedora.info/definitions/v4/repository#mixinTypes"))
```
Comment (acoburn) [inline fcrepo-http-api/src/main/resources/views/common.vsl]: mixinTypes? wasn't that removed?

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/ViewHelpers.java [hunk 2]
```diff
@@ -399,7 +399,17 @@ public Resource rdfsClass() {
      * @return content-bearing node for the given subject
      */
     public static Node getContentNode(final Node subject) {
-        return subject == null ? null : createURI(subject.getURI().replace("/" + FCR_METADATA, ""));
+        return subject == null ? null : NodeFactory.createURI(subject.getURI().replace("/" + FCR_METADATA, ""));
```
Comment (acoburn) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/ViewHelpers.java]: why not just use the static method below: `createURI(...)`?

**Comment by acoburn** [comment]
I don't even pretend to understand how the velocity templates work, so I don't know why `$nodeany` is changed to `$helpers`, but if that's what's needed, then sure, this looks good.

**Comment by ajs6f** [comment]
I'm perfectly willing to believe that what @awoods is doing here is good and needed. But I'll be walking away with the memory that one of our most thorough and thoroughly engaged committers, @acoburn, could only write "I don't even pretend to understand how the velocity templates work".

We should probably fix the problem that gave him the "opportunity" to write that.

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/ViewHelpers.java (hunk 2 above)
Comment (ajs6f) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/ViewHelpers.java]: It seems like the content of this change is exactly the opposite of what would seem to be a very natural suggestion from @acoburn. I wouldn't be all that surprised if @awoods had some good reason to do this, but the very whimsicality of our arrangements here bespeaks an ominous lack of gravity in their foundations.

File: fcrepo-http-api/src/main/resources/views/common.vsl (hunk 1 above)
Comment (awoods) [inline fcrepo-http-api/src/main/resources/views/common.vsl]: Although it does not appear that this macro is being used, I wanted to make sure that it at least worked.
https://jena.apache.org/documentation/javadoc/jena/org/apache/jena/graph/Node_ANY.html no longer has the `createURI` method in Jena 3.

Comment (acoburn) [inline fcrepo-http-api/src/main/resources/views/common.vsl]: So perhaps a separate PR could remove that block?

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/ViewHelpers.java (hunk 2 above)
Comment (awoods) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/responses/ViewHelpers.java]: @acoburn, I chose to use `NodeFactory.createURI` as a direct replacement for the previous `Node_ANY.createURI` in order to minimize the change in logic as well as keep the `getContentNode` and local `createURI` methods decoupled.
But, as you observed, the functionality is the same either way.
If there is a good reason to do so, I would be fine with changing this to use the local `createURI`.

**Comment by awoods** [comment]
The root to this PR is changing from using the now non-existent:
https://jena.apache.org/documentation/javadoc/jena/org/apache/jena/graph/Node_ANY.html - `createURI`
to using:
https://jena.apache.org/documentation/javadoc/jena/org/apache/jena/graph/NodeFactory.html#createURI-java.lang.String-

File: fcrepo-http-api/src/main/resources/views/common.vsl (hunk 1 above)
Comment (awoods) [inline fcrepo-http-api/src/main/resources/views/common.vsl]: Absolutely.
https://jira.duraspace.org/browse/FCREPO-2125

Comment (ajs6f) [inline fcrepo-http-api/src/main/resources/views/common.vsl]: We could remove [a lot more](https://jira.duraspace.org/browse/FCREPO-1594). :)

**Comment by awoods** [comment]
Is there any reason not to merge this? 
I do not believe there are any outstanding comments.

**Pull request merged**